073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,250 @@
+
+### Example flows
+
+When a new cluster is first created the flow is:
+
+1. Either user or CaReconciler creates the Cluster and Client CA certificates and keys and stores them in Kubernetes Secrets
+2. The CaReconciler generates the Cluster Operator certificate and stores it in the <CLUSTER_NAME>-cluster-operator-certs Kubernetes Secret
+3. The component reconcilers (KafkaReconciler, ZooKeeperReconciler, EntityOperatorReconciler etc) generate their certificates, store them in Kubernetes Secrets and annotate the Kubernetes Secrets and Pods with the cert and key generations
+
+When the CA private key is updated (either because it has expired or the user has incremented the generation to indicate they've manually updated the Kubernetes Secret):
+1. If Strimzi is managing the CA, the CaReconciler renews the Cluster and Client CAs certificates and keys. The old certificate and key are kept in the Kubernetes Secret and renamed to have the key ca-YYYY-MM-DDTHH-MM-SSZ.crt.
+2. The CaReconciler skips updating the <CLUSTER_NAME>-cluster-operator-certs Kubernetes Secret (since the components still only trust the old CA)
+3. The CaReconciler rolls the ZooKeeper and Kafka nodes to trust the new CA certificates
+4. The ZooKeeper and Kafka reconcilers generate new certificates signed by the new CA key and roll the pods to use the new certificates
+5. The EntityOperator, CruiseControl and KafkaExporter reconcilers generate new certificates signed by the new CA key amd roll the pods to use the new certificates
+6. A new reconciliation starts
+7. The CaReconciler updates the <CLUSTER_NAME>-cluster-operator-certs Kubernetes Secret with the new CA certificate (since the generations are now correct)
+8. The CaReconciler removes the old CA certificates and keys that have been named ca-YYYY-MM-DDTHH-MM-SSZ.crt.
+
+The existing logic assumes two things:
+ - all certificates used by components have been signed by the same root CA certificate
+ - the CaReconciler runs and (if required) has either created or updated the CA certificates and keys before the other reconcilers run, i.e. the CA certificate is already present when e.g. the KafkaReconciler starts
+
+## Motivation
+
+In future it would be beneficial if Strimzi can integrate with commonly deployed certificate management solutions, such as [Cert Manager][cmio], or [Vault][vault].
+These solutions work by having the user request certificates from a particular issuer.
+
+Integrating with these solutions requires a change in how things are managed. Specifically:
+ - Strimzi cannot guarantee that the issuer will issue every certificate signed by the same root CA certificate. 
+   - For example Cert Manager will include the CA certificate to trust when it issues the certificate (if the CA certificate is not a public one)
+   - Using those solutions therefore requires TLS peers to be able to trust multiple root CA certificates.
```

Comment:
I still think this is heavily overstated. The Cert Manager-based workflow will make you trust a single CA. That might require handling of multiple trusted certs during renewals or moves, but I do not think it justifies the whole apparatus built below.